### DISCUSSION TITLE
Thought about migrating codebase to Swift and Kotlin

### DISCUSSION THREAD

**Discussion by hyochan** [hyochan/react-native-iap#1417] [Ideas]
I want to know other people's thoughts on migrating codebase to new languages.

Since android prefers `Kotlin` more and Apple is pushing `Swift`, I think it will help future maintenance when migrating. 

Please share your thoughts :pray:

**Comment by andresesfm** [comment]
Here is the first draft of the migration to swift. Still WIP: https://github.com/dooboolab/react-native-iap/pull/1466

**Comment by andresesfm** [comment]
@hyochan I wanted to start the discussion of releasing the above PR as an alpha-1 build (react-native-iap:8.0-alpha-1) in order to get people to help test more easily. But it's definitely not ready to be merged into master. What are your thoughts on how to proceed?
